Tests/StarknetTests/Accounts/AccountTest.swift

```diff
@@ -130,6 +168,31 @@ final class AccountTests: XCTestCase {
         XCTAssertEqual(newNonce.value - nonce.value, Felt.one.value)
     }
 
+    func testDeployAccountV3() async throws {
+        let newSigner = StarkCurveSigner(privateKey: 4567)!
+        let newPublicKey = newSigner.publicKey
+        let newAccountAddress = StarknetContractAddressCalculator.calculateFrom(classHash: accountContractClassHash, calldata: [newPublicKey], salt: .zero)
+        let newAccount = StarknetAccount(address: newAccountAddress, signer: newSigner, provider: provider, cairoVersion: .zero)
+
+        try await Self.devnetClient.prefundAccount(address: newAccountAddress, unit: .fri)
+
+        let nonce = await (try? newAccount.getNonce()) ?? .zero
+
+        let feeEstimate = try await newAccount.estimateDeployAccountFeeV3(classHash: accountContractClassHash, calldata: [newPublicKey], salt: .zero, nonce: nonce)
+
+        let params = StarknetExecutionParamsV3(nonce: nonce, l1ResourceBounds: feeEstimate.toResourceBounds().l1Gas)
```

Comment:
This `l1ResourceBounds: feeEstimate.toResourceBounds().l1Gas` seems not very intuitive. Should we make `StarknetExecutionParamsV3` initializer accept  full resource bounds mapping instead?
Then, it would look like that instead:
```swift
let params = StarknetExecutionParamsV3(nonce: nonce, resourceBounds: feeEstimate.toResourceBounds())
```

But that would make the use case with manually set resource bounds less intuitive either. We create object only to store a singe property.
```swift
let params = StarknetExecutionParamsV3(nonce: nonce, resourceBounds: StarknetResourceBoundsMapping(l1Gas: StarknetResourceBounds(maxAmount: 1000, maxPricePerUnit: 1234)))
```

We can also have both constructors, and that too could be confusing.

Reply:
I'd leave it as it is right now to be honest, it may not be very intuitive, but is more explicit